consensus/src/runway/backup.rs

```diff
@@ -0,0 +1,331 @@
+use crate::{units::UncheckedSignedUnit, Data, Hasher, Round, Sender, Signature};
+use codec::{Decode, Encode, Error as CodecError};
+use futures::channel::oneshot;
+use log::{error, warn};
+use std::io::{Read, Write};
+
+#[derive(Debug)]
+/// Backup load error. Could be either caused by io error from Reader, or by decoding.
+pub enum LoaderError {
+    IO(std::io::Error),
+    Codec(CodecError),
+}
+
+impl From<std::io::Error> for LoaderError {
+    fn from(err: std::io::Error) -> Self {
+        Self::IO(err)
+    }
+}
+
+impl From<CodecError> for LoaderError {
+    fn from(err: CodecError) -> Self {
+        Self::Codec(err)
+    }
+}
+
+/// Abstraction over Unit backup saving mechanism
+pub trait UnitSaver<H: Hasher, D: Data, S: Signature> {
```

Comment:
I don't think these need to be traits. You can just have them as structs which take a `W` and `R` and then implement `Write` and `Read` for the sake of tests.

Reply:
Both UnitSaver and UnitLoader don't need any internal state, just the methods provided by `W` and `R`. So I thought creating traits that have a default implementation for each `std::io::Write/Read` would be better. Otherwise i would have to create structs with only `inner: W/R`. Do you think such approach is better? I'm not sure what you mean by "implement Write and Read for the sake of tests".

Reply:
I changed it to have `UnitSaver/Loader<W/R>` and save/load methods are dependent on `<H,D,S>` do you think this is enough type safety or should I also parameterize `UnitSaver/Loader<W/R, H, D, S>` so that only one type of unit can be saved/loaded?

Reply:
Hmm, that's a good point, I don't think it's strictly necessary, but I support doing this.